### PR TITLE
Genres episodes

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.12.0
- Added: Extracting Genre and Season/Episode numbers for EPG entries and recordings
- Added: Updated Readme
- Fixed: Minimum version now openwebif 1.3.0 to use autotimers

### CODE REVIEW

File: src/Timers.cpp [hunk 1]
```diff
@@ -563,10 +565,10 @@ void Timers::GetTimerTypes(std::vector<PVR_TIMER_TYPE> &types) const
   {
     TimerType(unsigned int id, unsigned int attributes,
       const std::string &description = std::string(),
-      const std::vector< std::pair<int, std::string> > &groupValues
-        = std::vector< std::pair<int, std::string> >(),
-      const std::vector< std::pair<int, std::string> > &deDupValues
-        = std::vector< std::pair<int, std::string> >())
+      const std::vector<std::pair<int, std::string>> &groupValues
+        = std::vector<std::pair<int, std::string>>(),
+      const std::vector<std::pair<int, std::string> > &deDupValues
```
[the remark this answers is not in the document]
Comment (phunkyfish) [reply src/Timers.cpp]: Done

File: src/extract/GenreExtractor.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
+#include "GenreExtractor.h"
+#include <regex>
+
+using namespace vuplus;
+using namespace ADDON;
+
+GenreExtractor::GenreExtractor()
+{
+  for (std::map<int, std::string>::const_iterator it = kodiKeyToGenreMap.begin(); it != kodiKeyToGenreMap.end(); it++)
```
[the remark this answers is not in the document]
Comment (phunkyfish) [reply src/extract/GenreExtractor.cpp]: Done

File: src/extract/EpgEntryExtractor.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#include "GenreExtractor.h"
+#include "ShowInfoExtractor.h"
+
+using namespace vuplus;
+using namespace ADDON;
+
+EpgEntryExtractor::EpgEntryExtractor()
+{
+   extractors.emplace_back(new GenreExtractor());
+   extractors.emplace_back(new ShowInfoExtractor());
```
[the remark this answers is not in the document]
Comment (phunkyfish) [reply src/extract/EpgEntryExtractor.cpp]: The vector is using unique_pointer: std::vector<std::unique_ptr<IExtractor>> extractors;

I've never used them before but my understanding was the deletion was automatic.

[the remark this answers is not in the document]
Comment (ksooo) [reply src/extract/EpgEntryExtractor.cpp]: Ah okay, overlooked this. Sorry.

**Pull request merged**